lib/nice_partials/partial.rb

```diff
@@ -32,6 +32,27 @@ def helpers(&block)
       class_eval &block
     end
 
+    # Allows an inner partial to extract content from an outer partial.
+    #
+    # Additionally a hash of keys to rename in the new partial context can be passed.
+    #
+    #   First, an outer partial gets some content set:
+    #   <% partial.content_for :title, "Hello there" %>
+    #   <% partial.content_for :byline, "Somebody" %>
+    #
+    #   Second, a new partial is rendered, but we want to extract the title, byline content but rename the byline key too:
+    #   <%= render "shared/title" do |cp| %>
+    #     <% cp.extract_from partial, :title, byline: :name %>
+    #   <% end %>
+    #
+    #   # Third, the contents with any renames are accessible in shared/_title.html.erb:
+    #   <%= partial.content_for :title %> # => "Hello there"
+    #   <%= partial.content_for :name %> # => "Somebody"
+    def extract_from(partial, *names, **renames)
+      names.chain(renames).each { |key, new_key = key| content_for new_key, partial.content_for(key) }
```

Comment:
Actually if we want this to be `partial.relay cp, :title, byline: :name`, we'd need to change this:

```suggestion
    def relay(partial, *names, **renames)
      names.chain(renames).each { |key, new_key = key| partial.content_for new_key, content_for(key) }
```

Hm, I can't decide and I've confused myself with why I went for `extract_from` — `relay` clarifies the key renaming aspect a bit, although it's technically outside the outer partial's concern to know what an inner context needs.